gradient_boosting.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+
+from sklearn.base import RegressorMixin
+from tree import DecisionTreeRegressor
+
+
+class BaseGradientBoostingRegressor:
+
+    def __init__(self,
+                 n_estimators: 5,
+                 max_depth: int,
+                 learning_rate=None):
+
+        self.estimator = DecisionTreeRegressor
```

Comment:
You don't use variable `estimator` outside the class. Please, make it protected via a single underscore (`self._estimator`).